core/shared/src/main/scala/scalaz/zio/duration/Duration.scala

```diff
@@ -0,0 +1,73 @@
+package scalaz.zio.duration
+
+import java.util.concurrent.TimeUnit
+import scala.concurrent.duration.{ Duration => ScalaDuration, FiniteDuration => ScalaFiniteDuration }
+import scala.math.Ordered
+
+sealed trait Duration extends Ordered[Duration] {
+
+  def +(other: Duration): Duration
+
+  def *(factor: Double): Duration
+
+  def max(other: Duration): Duration = if (this > other) this else other
+
+  def min(other: Duration): Duration = if (this < other) this else other
+
```

Comment:
To avoid the need for pattern matching on type, let's add a fold method here:

```scala
final def fold[Z](infinity: => Z, finite: Finite => Z): Z = ...
```

Then we can use this `fold` method instead of pattern matching.